sims/networks/basic/timing/GunnsRosesTiming.cpp

```diff
@@ -79,34 +79,40 @@ void GunnsRosesTiming::update(const double timestep)
 {
     if (N <= N_STOP) {
         if (0 == step) {
-            if (network) delete network;
+            delete network;  // No need to check for null
+
+            // Create a network with a unique_ptr
             std::ostringstream name;
             name << "rosesBenchmark.network_" << N;
+
             if (Gunns::GPU_SPARSE == gpuMode) {
                 name << "_gpu_sparse";
             } else if (Gunns::GPU_DENSE == gpuMode) {
                 name << "_gpu";
-            } else if (Gunns::NO_GPU == gpuMode) {
+            } else {
                 name << "_cpu";
             }
             std::string nameString = name.str();
             network = new TimingBasic(nameString);
             network->N = N;
             network->initialize(nameString);
             network->netSolver.setGpuOptions(gpuMode, 1);
-            if (islands) network->netSolver.setIslandMode(Gunns::SOLVE);
+            if (islands) {
+                network->netSolver.setIslandMode(Gunns::SOLVE);
+            }
         }
+        
         network->update(timestep);
         step++;
+        
         if (N_STEPS <= step) {
             storePotentials();
-//            dumpA();
```

Comment:
I want to keep these 2 'dump' calls in, and commented out, so that we can uncomment them for debugging.  Please restore these lines and resubmit.  If you want, you can add a comment like "these are for debugging, uncomment as needed", etc.